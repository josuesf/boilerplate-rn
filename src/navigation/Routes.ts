import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {IProduct} from 'src/common/models/Products';

export const ROUTE_HOME = 'Home';
export const ROUTE_PROD_DETAIL = 'ProductDetail';

export type RootStackParamList = {
  Home: undefined;
  ProductDetail: {product: IProduct};
};
export type NavigationProp = NativeStackNavigationProp<
  RootStackParamList,
  'ProductDetail'
>;
