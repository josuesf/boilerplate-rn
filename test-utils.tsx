import React from 'react';
import {render as rntlRender} from '@testing-library/react-native';
import {ThemeProvider} from './@components';
import moment from 'moment';
import 'moment/locale/es-mx';
import {Provider} from 'react-redux';
import {store} from 'src/redux/store';

function render(ui: React.ReactElement) {
  moment.locale('es-mx');
  return rntlRender(ui, {
    wrapper: ({children}) => (
      <ThemeProvider>
        <Provider store={store}>{children}</Provider>
      </ThemeProvider>
    ),
  });
}

export * from '@testing-library/react-native';

export {render};
