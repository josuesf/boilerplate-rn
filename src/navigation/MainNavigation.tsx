import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import * as React from 'react';
import {HomeScreen} from 'src/scenes/Home/screens/HomeScreen';
import {ProductDetail} from 'src/scenes/Home/screens/ProductDetail';
import {RootStackParamList, ROUTE_HOME, ROUTE_PROD_DETAIL} from './Routes';

const Stack = createNativeStackNavigator<RootStackParamList>();

export const MainNavigation = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name={ROUTE_HOME}
          options={{headerShown: false}}
          component={HomeScreen}
        />
        <Stack.Screen
          name={ROUTE_PROD_DETAIL}
          options={{headerShown: false}}
          component={ProductDetail}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};
