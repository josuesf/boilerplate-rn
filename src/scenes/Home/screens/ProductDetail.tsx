import * as React from 'react';
import {Box, Button, Text} from '@components';
import {RFValue} from 'react-native-responsive-fontsize';
import {RootStackParamList} from 'src/navigation/Routes';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {useTranslation} from 'react-i18next';
import {FormatDate} from 'src/common/utils/Formats';
import {ImageProduct} from '../components/ImageProduct';
import {ScrollView} from 'react-native';

type Props = NativeStackScreenProps<RootStackParamList, 'ProductDetail'>;

export function ProductDetail({route, navigation}: Props) {
  const {product} = route.params;
  const {t} = useTranslation();
  const dateCreated = FormatDate(new Date(product.createdAt));
  return (
    <Box flex={1}>
      <ScrollView>
        <Box
          bg="purpleLight"
          justifyContent="flex-end"
          height={RFValue(150)}
          pl="m"
          pb="m">
          <Text variant="h2">{product.product}</Text>
        </Box>
        <Box p="m">
          <ImageProduct uri={product.image} />
          <Text variant="h4">{t('details.detailsProduct')}</Text>
          <Text my="m" variant="h3">{`Comprado el ${dateCreated}`}</Text>
          <Text variant="h4">{t('details.pointsEarned')}</Text>
          <Text my="m" variant="h3">{`${product.points} ${t('points')}`}</Text>
          <Button
            onPress={() => navigation.goBack()}
            variant="primary"
            label={t('accept')}
          />
        </Box>
      </ScrollView>
    </Box>
  );
}
