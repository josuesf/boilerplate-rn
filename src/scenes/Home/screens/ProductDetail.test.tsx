import React from 'react';
import {ProductDetail} from './ProductDetail';
import {render, screen} from 'test-utils';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from 'src/navigation/Routes';

type NavigationScreenPropAlias = NativeStackScreenProps<
  RootStackParamList,
  'ProductDetail'
>;
let nameScreen: 'ProductDetail';
const productSelected = {
  createdAt: '2022-12-09T10:20:00.909Z',
  product: 'Fantastic Granite Bacon',
  points: 200,
  image: 'https://loremflickr.com/640/480/technics',
  is_redemption: false,
  id: '3',
};
const ProductDetailProps = (): NavigationScreenPropAlias => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  route: {
    params: {
      product: productSelected,
    },
    key: '1',
    name: nameScreen,
  },
});

describe('ProductDetail', function () {
  test('Display correctly product selected', async () => {
    const props: NavigationScreenPropAlias = ProductDetailProps();
    const {getByTestId} = render(<ProductDetail {...props} />);
    expect(getByTestId('name-product')).toHaveTextContent(
      productSelected.product,
    );
    expect(getByTestId('date-created')).toHaveTextContent(
      'details.boughtWhere 9 de diciembre, 2022',
    );
    expect(getByTestId('points-earned')).toHaveTextContent(
      productSelected.points.toString(),
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
