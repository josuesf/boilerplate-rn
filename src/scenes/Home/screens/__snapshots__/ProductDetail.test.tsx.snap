// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail Display correctly product selected 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#CFD6FF",
              "height": 294,
              "justifyContent": "flex-end",
              "paddingBottom": 31,
              "paddingLeft": 31,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#0B0B0B",
                "fontFamily": "Avenir",
                "fontSize": 39,
                "fontWeight": "900",
                "lineHeight": 47,
              },
            ]
          }
          testID="name-product"
        >
          Fantastic Granite Bacon
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "padding": 31,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#9B9898",
                "borderRadius": 20,
                "elevation": 5,
                "height": 667,
                "justifyContent": "center",
                "marginBottom": 31,
                "shadowColor": "#0B0B0B",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.8,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://loremflickr.com/640/480/technics",
              }
            }
            style={
              Object {
                "borderRadius": 20,
                "height": 687,
              }
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#9B9898",
                "fontFamily": "Avenir",
                "fontSize": 27,
                "fontWeight": "900",
                "lineHeight": 37,
              },
            ]
          }
        >
          details.detailsProduct
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#0B0B0B",
                "fontFamily": "Avenir",
                "fontSize": 31,
                "fontWeight": "900",
                "lineHeight": 43,
                "marginVertical": 31,
              },
            ]
          }
          testID="date-created"
        >
          details.boughtWhere 9 de diciembre, 2022
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#9B9898",
                "fontFamily": "Avenir",
                "fontSize": 27,
                "fontWeight": "900",
                "lineHeight": 37,
              },
            ]
          }
        >
          details.pointsEarned
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#0B0B0B",
                "fontFamily": "Avenir",
                "fontSize": 31,
                "fontWeight": "900",
                "lineHeight": 43,
                "marginVertical": 31,
              },
            ]
          }
          testID="points-earned"
        >
          200 points
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#334FFA",
              "borderRadius": 20,
              "height": 98,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Avenir",
                  "fontSize": 31,
                  "fontWeight": "900",
                  "lineHeight": 39,
                },
              ]
            }
          >
            accept
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
